packages/core/src/amazonqGumby/chat/controller/controller.ts

```diff
@@ -327,6 +320,29 @@ export class GumbyController {
                 const pathToJavaHome = extractPath(data.message)
 
                 if (pathToJavaHome) {
+                    transformByQState.setJavaHome(pathToJavaHome)
+                    getLogger().info(
+                        `CodeTransformation: using JAVA_HOME = ${transformByQState.getJavaHome()} since source JDK does not match Maven JDK`
+                    )
+
+                    try {
+                        // When user provided a
+                        await validateCanCompileProject()
+                    } catch (err: any) {
+                        if (err instanceof JavaHomeNotSetError) {
+                            transformByQState.unsetJavaHome()
+                            this.sessionStorage.getSession().conversationState = ConversationState.PROMPT_JAVA_HOME
+                            this.messenger.sendStaticTextResponse('java-home-not-set', data.tabID)
+                            this.messenger.sendChatInputEnabled(data.tabID, true)
+                            this.messenger.sendUpdatePlaceholder(
+                                data.tabID,
+                                `Incompatible JDK, Enter the path to your Java ${transformByQState.getSourceJDKVersion()} installation.`
```

Comment:
1. This is an update to the string and should be reviewed 
2. Was this interaction reviewed with Ishaani? I'm unsure if updating the placeholder to indicate the issue is the best way to do this, if only because it's not a pattern we repeat anywhere else with errors. 
3. Can we extract any of this to a common method shared between `processHumanChatMessage` and `validateBuildWithPromptOnError`? Can the former call the latter?
4. Does this retry need to be limited in any way, or is it ok that it's effectively an infinite retry loop?

Reply:
Great points, ill sync this with Ishaani, think that will answer 1,2,4. 

3. Can look into that

Reply:
Should add a retry limit of 3 and review with UX after.